backends/dpdk/tdiConf.cpp

```diff
@@ -0,0 +1,104 @@
+/* Copyright 2023 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "backends/dpdk/tdiConf.h"
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+
+namespace DPDK {
+
+void TdiBfrtConf::generate(DPDK::DpdkOptions &options) {
+    if (options.outputFile.isNullOrEmpty()) {
+        ::error(ErrorType::ERR_UNEXPECTED,
+                "No output file provided. Unable to generate correct TDI builder config file.");
+        return;
+    }
+    auto inputFile = std::filesystem::absolute(options.file.c_str());
+    auto outFile = std::filesystem::absolute(options.outputFile.c_str());
+    auto outDir = outFile.parent_path();
+    auto tdiFile = std::filesystem::absolute(options.tdiBuilderConf.c_str());
+    auto programName = inputFile.stem();
+
+    if (options.bfRtSchema.isNullOrEmpty()) {
+        options.bfRtSchema = (outDir / programName).replace_filename("json").c_str();
+        ::warning(
+            "BF-Runtime Schema file name not provided, but is required for the TDI builder "
+            "configuration. Generating file %1%",
+            options.bfRtSchema);
+    }
+    if (options.ctxtFile.isNullOrEmpty()) {
+        options.ctxtFile = (outDir / "context.json").c_str();
+        ::warning(
+            "DPDK context file name not provided, but is required for the TDI builder "
+            "configuration. Generating file %1%",
+            options.ctxtFile);
+    }
```

Comment:
So the behavior here is that if you give the command line option to generate the TDI pipeline builder conf file, and also the command line options to specify file names for bfrt scheme and context json files, it will use those file names in the generated conf file?

And if you do not specify the options on the command line to generate those files, but do give the option to generate the TDI pipeline builder conf file, then it will issue warnings, and make up its own default file names for those other output files, and generate them anyway?

If so, that seems reasonable to me.

Reply:
> So the behavior here is that if you give the command line option to generate the TDI pipeline builder conf file, and also the command line options to specify file names for bfrt scheme and context json files, it will use those file names in the generated conf file?
> 
> And if you do not specify the options on the command line to generate those files, but do give the option to generate the TDI pipeline builder conf file, then it will issue warnings, and make up its own default file names for those other output files, and generate them anyway?
> 
> If so, that seems reasonable to me.

Yes. I considered throwing an error, but it seemed more convenient to generate all the files in one go with one parameter.